### PR TITLE
WritePrepared Txn: add stats

### PR DESCRIPTION
**Pull request by maysamyabandeh**
Adding some stats that would be helpful to monitor if the DB has gone to unlikely stats that would hurt the performance. These are mostly when we end up needing to acquire a mutex.

### CODE REVIEW

File: include/rocksdb/statistics.h [hunk 1]
```diff
@@ -304,6 +304,17 @@ enum Tickers : uint32_t {
   // # of bytes in the blob files evicted because of BlobDB is full.
   BLOB_DB_FIFO_BYTES_EVICTED,
 
+  // These coutners indicate a performance issue in WritePrepared transactions.
+  // We should not seem them ticking them much.
+  // # of times prepare_mutex_ is acquired in the fast path.
+  TXN_PREPARE_MUTEX_OVERHEAD,
```
Comment (yiwu-arbug) [inline include/rocksdb/statistics.h]: by "overhead" I thought it mean time spent, but they are just counts. Should we call them "xxx_count" instead?

Comment (maysamyabandeh) [reply include/rocksdb/statistics.h]: But we do not count it all the time; only when they are acquired on the fast path.

File: db/db_impl.h [hunk 2]
```diff
@@ -688,14 +688,20 @@ class DBImpl : public DB {
...
-  // batch which will be written to memtable later during the commit, and in
-  // WritePrepared it is guaranteed since it will be used only for WAL markers
-  // which will never be written to memtable.
-  // batch_cnt is expected to be non-zero in seq_per_batch mode and indicates
-  // the number of sub-patches. A sub-patch is a subset of the write batch that
-  // does not have duplicate keys.
+  // batch will still be skipped from memtable during the recovery. An excption
+  // to this is seq_per_batch_ mode, in which since each batch already takes one
+  // seq, it is ok for the batch to write to memtable during recovery as long as
+  // it only takes one sequence number: i.e., no duplicate keys.
```
[the remark this answers is not in the document]
Comment (maysamyabandeh) [reply db/db_impl.h]: Then during recovery the sequence number of key/values after the duplicate key matched with the sequence number of keys of the next batch; their fate for visibility then would be mixed with the visibility status of the next batch.
Late I am going to add harness code to recovery to detect such mistakes.

**Comment by maysamyabandeh** [comment]
The failure is unrelated and cannot produce it locally:
```
[ RUN      ] DBFlushTest.SyncFail
db/db_flush_test.cc:92: Failure
Value of: cfd->current()->TEST_refs()
  Actual: 2
Expected: refs_before
Which is: 3
terminate called after throwing an instance of 'testing::internal::GoogleTestFailureException'
  what():  db/db_flush_test.cc:92: Failure
Value of: cfd->current()->TEST_refs()
  Actual: 2
Expected: refs_before
Which is: 3
```

**Pull request closed without merging**